chore: update package type to "module"

cedoor opened the pull request:
## What is this PR doing?

This PR resolves compatibility issues with Node.js by setting the package type to module in package.json. This ensures the package works in both modern browsers and Node.js environments.

Related: https://github.com/voltrevo/summon-ts/pull/1

## How can these changes be manually tested?

It can be tested locally by trying to use this package from NodeJS.

## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

voltrevo reviewed (changes requested):
Couple things:

1. Not sure we can remove `fixEmscriptenCode`. I note that you have updated the options to export es6, but when I run the build I still see `require("fs")` outside a try/catch in there, which breaks in NextJS due to their static checking of `require`.

2. `npm run build` fails:

```
tests/secure2PC.test.ts:2:40 - error TS2307: Cannot find module '../' or its corresponding type declarations.

2 import { BufferQueue, secure2PC } from "../"
                                         ~~~~~
```

I think we can just put `../src/ts/` here instead.

cedoor commented:
@voltrevo 

1. I thought requires were replaced by imports, but it's still there you're right. reverted!
2. Updated!

I also added the `.js` extensions and tested it on NodeJS.

cedoor commented:
@voltrevo This code still needs some fixes.

Current state based on our conv:

After adding `{ type: 'module' }` as a second parameter of `new Worker`, I see the following error `Error: Module is not defined`, which is probably related to the `build/jslib.js` file (which is compatible with ES6 now).

cedoor commented:
@voltrevo Ready!

voltrevo reviewed (approved):
Excellent work!

I just added a change around concurrency checks:
- worker: Switched to an explicity concurrency check, previously we were using Module.emp as a global
- node: Removed concurrency check because it's not needed
